### ISSUE TITLE
Optional values with floats

### ISSUE THREAD

**Issue by theofidry** [nelmio/alice#494]
I found the following test:

```php
public function testLoadParsesOptionalValuesWithFloats()
    {
        $this->loadData([
            self::USER => [
                'user0' => [
                    'username' => '0.5? name',
                ],
                'user1' => [
                    'username' => '0.5? name : nothing',
                ],
                'user2' => [
                    'username' => '0? name : nothing',
                ],
                'user3' => [
                    'username' => '1? name : nothing',
                ],
            ],
        ]);

        $this->assertContains($this->loader->getReference('user0')->username, ['name', null]);
        $this->assertContains($this->loader->getReference('user1')->username, ['name', 'nothing']);
        $this->assertEquals('nothing', $this->loader->getReference('user2')->username);
        $this->assertEquals('name', $this->loader->getReference('user3')->username);
    }
```

From the look of it, instead of having a percentage like `80%? something : nothing`, you can instead use a float value i.e. `0.8? something : nothing`. I did not find any occurrence of this feature in the doc.

As this feature has not been documented and it is a bit redundant, what do you think of deprecating it @tshelburne?

**Comment by tshelburne** [comment]
I think we should deprecate - not only is it documented, but the intention is much less clear.

**Comment by theofidry** [comment]
Closed via #564
